casper/src/main/rholang/NonNegativeNumber.rho

```diff
@@ -47,8 +47,13 @@ new NonNegativeNumber, rs(`rho:registry:insertSigned:ed25519`), uriOut in {
         }
       } |
       return!(bundle+{*this}) |
-      if (init >= 0) { valueStore!(init) }
-      else           { valueStore!(0)    } //Initial balance is zero if given is negative}
+      match init { //Initial balance is zero if given is negative on non-integer
+        Int => {
```

Comment:
Did you consider making Int part of the contract pattern? i.e. `contract NonNegativeNumber(@(init /\ Int), return) = { ... }`